cleverhans/attacks_tf.py

```diff
@@ -417,3 +417,245 @@ def jacobian_augmentation(sess, x, X_sub_prev, Y_sub, grads, lmbda,
 
     # Return augmented training data (needs to be labeled afterwards)
     return X_sub
+
+
+class CarliniWagnerL2:
```

Comment:
Yeah, I should remove the second class. It was a class so that I could easily deal with graph caching, but now that generate_np is rewritten, I should clean this up.